packages/neuron-wallet/src/keys/key.ts

```diff
@@ -0,0 +1,73 @@
+import bip32 from 'bip32'
+import bip39 from 'bip39'
+import { Keystore, Child } from './keystore'
+
+export default class Key {
+  private keystore: Keystore
+
+  private mnemonic: string
+
+  constructor(keystore: Keystore, mnemonic: string) {
+    this.keystore = keystore
+    this.mnemonic = mnemonic
+  }
+
+  public static fromKeystore = (keystore: Keystore) => {
+    return new Key(keystore, '')
+  }
+
+  public static fromKeystoreJson = (json: string) => {
+    return new Key(JSON.parse(json), '')
+  }
+
+  getKeystore = () => this.keystore
+
+  getKeystoreJson = () => JSON.stringify(this.keystore)
+
+  getMnemonic = () => this.mnemonic
+
+  public static generateKey = () => {
+    const mnemonic = bip39.generateMnemonic()
+    return Key.fromMnemonic(mnemonic, false)
+  }
+
+  public static fromMnemonic = (mnemonic: string, derive: boolean) => {
+    const seed = bip39.mnemonicToSeed(mnemonic)
+    const root = bip32.fromSeed(seed)
+    const master = {
+      privateKey: root.privateKey.toString('hex'),
+      chainCode: root.chainCode.toString('hex'),
+    }
+    const keystore: Keystore = {
+      master,
+    }
+    if (derive) {
+      keystore.children = Key.searchUsedAddress()
+    }
+    return new Key(keystore, mnemonic)
+  }
+
+  private static searchUsedAddress() {
```

Comment:
* If the function returns a collection of addresses, it should use `searchUsedAddresses` instead of `searchUsedAddress`.
* Are `address` and `child` of the same concept? If not, then `address` in the name might not be appropriate.

Reply:
Yes, `Addresses` will be better. It's my fault. 

I think the public key(or address) of master and child is same concept. So address should contain master and child.

Reply:
No address is a much simpler concept. It is a `public key` (could be child key's) hash.

Reply:
Public key and address are  one to one correspondence, and they are all generated from private key which contains master private key and child private key. 

But I think address just is one part of master and child, So searchAddress is not appropriate. 

I will think twice.

Reply:
Let's merge and update later.